### PR TITLE
Off-chain module for liqwid-nix 2

### PR DESCRIPTION
**Pull request by groscoe**
Good to go, now. Some rough edges still outstanding:

* It would be nice to be able to declare more than one test suite, each with a different config.
* CTL's `launchSearchablePursDocs` unfortunately hardcodes a `src/` path for the off-chain code. This means that projects which use a different folder structure can't use the default app to launch and build the docs.
* Apps, checks, etc are prefixed with the project name, which is weird when there's only one ("default"). We could handle this case separately and automatically make the single project the "default".

### CODE REVIEW

File: nix/offchain.nix
Comment (SeungheonOh) [inline nix/offchain.nix:206]: I think `ctl-server` is deprecated in favor of `runtime`. We don't need it.

Comment (SeungheonOh) [reply nix/offchain.nix]: I thought having it would give the annoying 
`trace: Warning: 'ctl-server' has moved to 'overlays.ctl-server' and will be removed from 'overlays.runtime' soon` but yeah, I think it will be fine to keep it there.

Comment (groscoe) [reply nix/offchain.nix]: It seems like even not having it gives the annoying logs, unfortunately.

File: nix/offchain.nix
Comment (SeungheonOh) [inline nix/offchain.nix:144]: `extraConfig` can also be an overly-like function that takes the final value. https://github.com/Plutonomicon/cardano-transaction-lib/blob/a83dff3465f67c8d7b339bd5925a1440c5aab977/nix/runtime.nix#L104
This is quite useful and I think we need to allow setting extraConfig using a function

File: nix/offchain.nix
Comment (SeungheonOh) [inline nix/offchain.nix:416]: Can we add `nix run` commands? 

- formatting
- pursuit
- updating spago2nix(maybe)

few things I can think of

File: nix/offchain.nix
Comment (SeungheonOh) [inline nix/offchain.nix:457]: wrong name. should be `all_offchain`

**Review by SeungheonOh — CHANGES_REQUESTED** [review]
Looks great! I see few missing things, and one mislabeling.

File: nix/offchain.nix
Comment (groscoe) [reply nix/offchain.nix:457]: You're right, good catch.

File: nix/offchain.nix
Comment (groscoe) [reply nix/offchain.nix:144]: The problem with using fixpoints like that is they're usually a nightmare to debug when something goes wrong. Could you give me an example of a config that would benefit from this change?

File: nix/offchain.nix
Comment (groscoe) [reply nix/offchain.nix:416]: Sure thing, thanks for the suggestion.

Comment (groscoe) [reply nix/offchain.nix:416]: Note that we already have pursuit, though, with `nix run .#<project-name>_.docs.

File: nix/offchain.nix
Comment (SeungheonOh) [reply nix/offchain.nix:144]: If that's the case, it would be nice to add the wallet server option. AP is in the Liqwid-onchain repo.

Comment (SeungheonOh) [reply nix/offchain.nix:144]: https://github.com/Liqwid-Labs/liqwid-offchain/pull/222

Comment (groscoe) [reply nix/offchain.nix:144]: I'd propose holding off on that until we actually merge it to off-chain. We can then iterate on this flake adding more scripts, tools, etc according to what we're using.

Comment (SeungheonOh) [reply nix/offchain.nix:144]: Yeah, okay :+1:

File: nix/offchain.nix
Comment (groscoe) [reply nix/offchain.nix:416]: Done!

**Review by SeungheonOh — APPROVED** [review]
Awesome! Thank you! 

Just in case, it would be nice to get another review from @emiflake as I might've missed some wired nix quarks :)

**Review by emiflake — CHANGES_REQUESTED** [review]
It's also missing CHANGELOG entry.

Other than these two, it looks pretty good! I guess we don't fully know how to allow it to take dependencies, but this can be added later. Sorry it took a while to review.

**Review by emiflake — APPROVED** [review]
I'm happy with it now.

**Pull request merged**
